fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLBuilder.java

```diff
@@ -118,9 +121,34 @@ public String getSQLTemplate() {
     /*
      * Returns a String representation suitable for debugging and log output.
      * This is ONLY intended for debugging in logs, and NEVER for passing to a JDBC database.
+     */
     @Override
     public String toString() {
-        return "SQLBuilder{..."; // TODO implement this...
+        StringBuilder whereClause  = new StringBuilder();
+        ArrayList<String> modifiedArgs = new ArrayList<String>();
```

Comment:
you can probably avoid creating this intermediate list? Just iterate and append to the `StringBuilder`, no? Or perhaps I'm missing something here (quick review only; didn't read the code in full details).